### PR TITLE
feat(api): impl Clone and Debug for Device and Stream

### PR DESCRIPTION
**Pull request by minghuaw**
1. Added `Clone` impl for `Device`, `Stream`, and `StreamOrDevice`
2. Deny `missing_debug_implementations` (pub types that don't have `Debug` impl will give an compile error`)

### CODE REVIEW

File: src/device.rs [hunk 1]
```diff
@@ -55,6 +85,15 @@ impl Default for Device {
     }
 }
 
+impl std::fmt::Debug for Device {
```
Comment (dcvz) [inline src/device.rs:88]: Can we not just use the display impl here?

File: src/stream.rs [hunk 2]
```diff
@@ -102,6 +133,14 @@ impl Default for Stream {
     }
 }
 
+impl std::fmt::Debug for Stream {
```
Comment (dcvz) [inline src/stream.rs:136]: Same as debug for device

**Review by dcvz — APPROVED** [review]
LGTM 👍 some small comments

File: src/device.rs (hunk 1 above)
Comment (minghuaw) [reply src/device.rs:88]: I think this is more a rust convention thing. I read somewhere that a recommended practice for library crates is to always provide `Debug` impl, which is why I added the `#![deny(missing_debug_implementations)]`. And `Display` is considered rather optional, but it doesn't hurt to provide both

Comment (minghuaw) [reply src/device.rs:88]: My brain's rusted lol. Misread your comment. Yeah, we could, copilot just generated the same code and I was too lazy to change that

**Pull request merged**